### PR TITLE
Converted Missile TurboBoost from boolean to integer percentage

### PR DESCRIPTION
**Pull request by reaperrr**
A small step towards #7509.

### CODE REVIEW

File: OpenRA.Mods.Common/Effects/Missile.cs [hunk 1]
```diff
@@ -41,8 +41,8 @@ class MissileInfo : IProjectileInfo
 		public readonly int ROT = 5;
 		[Desc("Explode when following the target longer than this.")]
 		public readonly int RangeLimit = 0;
-		[Desc("If fired at aircraft, increase speed by 50%.")]
-		public readonly bool TurboBoost = false;
+		[Desc("If fired at aircraft, increase speed by this percentage.")]
```
Comment (penev92) [inline OpenRA.Mods.Common/Effects/Missile.cs]: "fired by an aircraft"
Also should be only for planes, not helicopters.

**Comment by penev92** [comment]
OK, so you have a description that specifically states that this is for aircraft, you change it, but it's still for aircraft, and then you add this to infantry and tanks.
I don't see why the TS missiles need TurboBoost, and I have some issues with that description.

**Comment by penev92** [comment]
My only problem with this is... why would the missile get a speed boost when it's fired at an aircraft?

**Comment by chrisforbes** [comment]
Do we even need TurboBoost? Why not just set up two weapons, with different
acceptable targets and different stats?

**Comment by penev92** [comment]
To be fair I feel that that approach is even more hacky.
But I was told on IRC that that's how we've already done it for a few unit types.

**Comment by Phrohdoh** [comment]
I agree with @chrisforbes on this.

**Comment by reaperrr** [comment]
> Do we even need TurboBoost? Why not just set up two weapons, with different acceptable targets and different stats?

It does save some yaml bloat over a duplicate weapon with different target(s) and speed, but other than that you're right, of course.

**Comment by Unit158** [comment]
I agree. It feels hard-coded and gross.

**Comment by reaperrr** [comment]
Closing this, I'll cook up something different.

**Comment by penev92** [comment]
I don't agree with closing this like this.
Either we remove TurboBoost completely, or we fix it. People don't seem to want a fix. Then it should just go.

**Comment by reaperrr** [comment]
> I don't agree with closing this like this.
Either we remove TurboBoost completely, or we fix it. People don't seem to want a fix. Then it should just go.

Agreed, but when I said 'I'll cook up something different' I meant that I've gotten some new ideas for the ```Missile``` code that go beyond just removing ```TurboBoost```.

**Comment by pchote** [comment]
> It does save some yaml bloat over a duplicate weapon with different target(s) and speed

If it doesn't already work then we should implement weapon inheritance to save on bloat in these cases.

**Pull request closed without merging**